src/test/scala/edu/ie3/mobsim/model/ChargingBehaviorSpec.scala

```diff
@@ -368,5 +369,44 @@ class ChargingBehaviorSpec extends UnitSpec with ChargingBehaviorTestData {
         UUID.fromString("7537c0b6-3137-4e30-8a95-db1c0f9d9b81") -> 5.0
       )
     }
+
+    "return the correct price rating" in {
+      val priceRating = PrivateMethod[Double](Symbol("priceRating"))
+
+      val evWithLowPriceMemory: ElectricVehicle = ev1
+      evWithLowPriceMemory.updateChargingPricesMemory(0.0)
+
+      val evWithHighPriceMemory: ElectricVehicle = ev1
+      evWithHighPriceMemory.updateChargingPricesMemory(0.0)
+      evWithHighPriceMemory.updateChargingPricesMemory(1.0)
+
+      val evOtherPriceMemory: ElectricVehicle = ev1
+      evOtherPriceMemory.updateChargingPricesMemory(0.5)
+      evOtherPriceMemory.updateChargingPricesMemory(0.7)
+      evOtherPriceMemory.updateChargingPricesMemory(0.3)
+
+      val cases = Table(
+        ("ev", "prices", "expectedRating"),
+        (ev1, Map.empty[UUID, Double], 1.0),
+        (ev1, Map(cs6.getUuid -> 0.0), 1.0),
+        (evWithLowPriceMemory, Map(cs6.getUuid -> 0.0), 0.5),
+        (evWithHighPriceMemory, Map(cs6.getUuid -> 0.0), 1.0),
+        (evOtherPriceMemory, Map(cs6.getUuid -> 0.6), 0.24999999999999997),
+        (
+          evOtherPriceMemory,
+          Map(cs6.getUuid -> 0.6, cs6.getUuid -> 0.4),
+          0.7499999999999999
+        )
+      )
+
+      forAll(cases) { (ev, prices, expectedRating) =>
+        val result = ChargingBehavior invokePrivate priceRating(
+          cs6,
+          ev,
+          prices
+        )
+        result shouldBe expectedRating
```

Comment:
Btw, you can test doubles with tolerance like this:

`assert(result === expectedRating +- 1E-6)`